src/solver/variable/economy/STStorageCashFlowByCluster.h

```diff
@@ -0,0 +1,305 @@
+        }
+        // Next variable
+        NextType::yearBegin(year, numSpace);
+    }
+
+    void yearEnd(unsigned int year, unsigned int numSpace)
+    {
+        for (unsigned int clusterIndex = 0; clusterIndex < nbClusters_; ++clusterIndex)
+        {
+            // Compute all statistics from hourly results for the current year (daily, weekly, monthly, ...)
+            pValuesForTheCurrentYear[numSpace][clusterIndex].computeStatisticsForTheCurrentYear();
+        }
+        // Next variable
+        NextType::yearEnd(year, numSpace);
+    }
+
+    void computeSummary(std::map<unsigned int, unsigned int>& numSpaceToYear,
+                        unsigned int nbYearsForCurrentSummary)
+    {
+        for (unsigned int numSpace = 0; numSpace < nbYearsForCurrentSummary; ++numSpace)
+        {
+            for (unsigned int clusterIndex = 0; clusterIndex < nbClusters_; ++clusterIndex)
+            {
+                // Merge all those values with the global results
+                AncestorType::pResults[clusterIndex].merge(numSpaceToYear[numSpace],
+                                                pValuesForTheCurrentYear[numSpace][clusterIndex]);
+            }
+        }
+
+        // Next variable
+        NextType::computeSummary(numSpaceToYear, nbYearsForCurrentSummary);
+    }
+
+    void hourBegin(unsigned int hourInTheYear)
+    {
+        // Next variable
+        NextType::hourBegin(hourInTheYear);
+    }
+
+    void hourForEachArea(State& state, unsigned int numSpace)
+    {
+        for (uint clusterIndex = 0; clusterIndex != state.area->shortTermStorage.count();
+             ++clusterIndex)
+        {
+            // ST storage injection for the current cluster and this hour
+            // CashFlow[h] = (withdrawal - injection) * MRG. PRICE
+            pValuesForTheCurrentYear[numSpace][clusterIndex].hour[state.hourInTheYear]
+            = (state.hourlyResults->ShortTermStorage[state.hourInTheWeek].withdrawal[clusterIndex]
+            - state.hourlyResults->ShortTermStorage[state.hourInTheWeek].injection[clusterIndex])
+            * (-state.hourlyResults->CoutsMarginauxHoraires[state.hourInTheWeek]);
+            // Note: The marginal price provided by the solver is negative (naming convention).
+        }
+
+        // Next variable
+        NextType::hourForEachArea(state, numSpace);
+    }
+
+    inline void buildDigest(SurveyResults& results, int digestLevel, int dataLevel) const
+    {
+        // Ask to build the digest to the next variable
+        NextType::buildDigest(results, digestLevel, dataLevel);
+    }
+
+    Antares::Memory::Stored<double>::ConstReturnType retrieveRawHourlyValuesForCurrentYear(
+      unsigned int column,
+      unsigned int numSpace) const
+    {
+        return pValuesForTheCurrentYear[numSpace][column].hour;
+    }
+
+    inline uint64_t memoryUsage() const
+    {
+        uint64_t r = (sizeof(IntermediateValues) * nbClusters_ + IntermediateValues::MemoryUsage())
```

Comment:
what `r` is for?

Reply:
It is a temporary variable that's used to compute the memory usage.